torch/csrc/dynamo/eval_frame.c

```diff
@@ -512,14 +528,14 @@ static void call_profiler_end_hook(PyObject* record) {
 
 // Return value: borrowed reference
 // Is either Py_None or a PyCodeObject
-static PyObject* lookup(CacheEntry* e, THP_EVAL_API_FRAME_OBJECT *frame, CacheEntry* prev, size_t index) {
+static PyObject* lookup(CacheEntry* e, THP_EVAL_API_FRAME_OBJECT *frame, CacheEntry* prev, size_t index, PyObject** code_part) {
   if (e == NULL) {
     // NB: intentionally not using Py_RETURN_NONE, to return borrowed ref
     return Py_None;
   }
   PyObject *f_locals = frame->f_locals;
-  PyObject* valid = PyObject_CallOneArg(e->check_fn, f_locals);
-  if (unlikely(valid == NULL)) {
+  PyObject* result = PyObject_CallOneArg(e->check_fn, f_locals);
```

Comment:
A more elegant design may be to have check fn return a truthy object, that can have more information attached to the failure case than just False.

Reply:
Hmm, the failure info is already on the code part.

Reply:
Yes, but by returning a tuple you now have to unpack it and deal with the extra tuple allocation.